From c75b3fb8018edf6c955973ff3ec10647bf2cf3dc Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:30:03 -0400
Subject: [misc] genwqe: Fix potential memory leak when pinning memory

Message-id: <1413239416-7778-16-git-send-email-gduarte@redhat.com>
Patchwork-id: 97545
O-Subject: [RHEL7.1 PATCH BZ 1088491 15/28 v2] misc: genwqe: Fix potential memory leak when pinning memory
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: d913c7439add288b50752186b306634df2ae21e7

commit d913c7439add288b50752186b306634df2ae21e7
Author: Christian Engelmayer <cengelma@gmx.at>
Date:   Sat Jan 25 22:47:44 2014 +0100

    misc: genwqe: Fix potential memory leak when pinning memory

    Fix a memory leak in the genwqe_pin_mem() error path as called by
    ioctl GENWQE_PIN_MEM. In case there is an error encountered when
    mapping memory, the already allocated dma_mapping struct needs to
    be freed correctly.

    Detected by Coverity: CID 1162606.

    Signed-off-by: Christian Engelmayer <cengelma@gmx.at>
    Acked-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_dev.c b/drivers/misc/genwqe/card_dev.c
index 8f8a6b3..2c2c9cc 100644
--- a/drivers/misc/genwqe/card_dev.c
+++ b/drivers/misc/genwqe/card_dev.c
@@ -787,6 +787,7 @@ static int genwqe_pin_mem(struct genwqe_file *cfile, struct genwqe_mem *m)
  if (rc != 0) {
   dev_err(&pci_dev->dev,
    "[%s] genwqe_user_vmap rc=%d\n", __func__, rc);
+  kfree(dma_map);
   return rc;
  }
 
-- 
1.7.1